SamsungPay/src/main/java/com/braintreepayments/api/SamsungPayInternalClient.java

```diff
@@ -0,0 +1,164 @@
+import static com.samsung.android.sdk.samsungpay.v2.SpaySdk.SPAY_NOT_SUPPORTED;
+import static com.samsung.android.sdk.samsungpay.v2.SpaySdk.SPAY_READY;
+
+class SamsungPayInternalClient {
+
+    private final SamsungPay samsungPay;
+    private final PaymentManager paymentManager;
+
+    private final Set<SpaySdk.Brand> braintreeAcceptedCardBrands;
+
+    SamsungPayInternalClient(BraintreeClient braintreeClient, Configuration configuration) {
+        this(braintreeClient.getApplicationContext(), configuration, new SamsungPayPartnerInfoBuilder()
+                .setConfiguration(configuration)
+                .setSessionId(braintreeClient.getSessionId())
+                .setIntegrationType(braintreeClient.getIntegrationType())
+                .build());
+    }
+
+    private SamsungPayInternalClient(Context context, Configuration configuration, PartnerInfo partnerInfo) {
+        this(configuration, new SamsungPay(context, partnerInfo), new PaymentManager(context, partnerInfo));
+    }
+
+    @VisibleForTesting
+    SamsungPayInternalClient(Configuration configuration, SamsungPay samsungPay, PaymentManager paymentManager) {
+        this.braintreeAcceptedCardBrands = mapToSamsungPayCardBrands(configuration.getSupportedCardTypes());
+        this.samsungPay = samsungPay;
+        this.paymentManager = paymentManager;
+    }
+
+    void goToSamsungPayUpdatePage() {
+        samsungPay.goToUpdatePage();
+    }
+
+    void activateSamsungPay() {
+        samsungPay.activateSamsungPay();
+    }
+
+    void updateCustomSheet(CustomSheet customSheet) {
+        paymentManager.updateSheet(customSheet);
+    }
+
+    void getSamsungPayStatus(final GetSamsungPayStatusCallback callback) {
+        samsungPay.getSamsungPayStatus(new StatusListener() {
+            @Override
+            public void onSuccess(int statusCode, Bundle bundle) {
+                Exception samsungPayError = null;
+
+                @SamsungPayError int error = SamsungPayError.SAMSUNG_PAY_ERROR_UNKNOWN;
+                if (statusCode != SPAY_READY) {
+                    if (bundle != null && bundle.containsKey(SamsungPay.EXTRA_ERROR_REASON)) {
+                        int reason = bundle.getInt(SamsungPay.EXTRA_ERROR_REASON);
+                        switch (reason) {
+                            case ERROR_SPAY_APP_NEED_TO_UPDATE:
+                                error = SamsungPayError.SAMSUNG_PAY_APP_NEEDS_UPDATE;
+                                break;
+                            case ERROR_SPAY_SETUP_NOT_COMPLETED:
+                                error = SamsungPayError.SAMSUNG_PAY_SETUP_NOT_COMPLETED;
+                                break;
+                        }
+                    } else {
+                        if (statusCode == SPAY_NOT_READY) {
+                            error = SamsungPayError.SAMSUNG_PAY_NOT_READY;
+                        } else {
+                            // SPAY_NOT_SUPPORTED
```

Comment:
Couldn't this also be an unknown error? Might be safer to assert on the actual error code here too

Reply:
Yeah that makes sense. Technically there are 3 possible values for get status, but it'll be good to be defensive here just in case.

Reply:
Done!